src/AzureBlobStorageUpdateSession/UpdateSession/UpdateSessionStatus.cs

```diff
@@ -0,0 +1,19 @@
+using System.Collections.Generic;
+
+namespace Etg.Yams.Azure.UpdateSession
+{
+    public class UpdateSessionStatus
+    {
+        public IEnumerable<UpdateDomainEntity> InstancesEntities { get; }
```

Comment:
If you made this a List or even an array, you'll get a count for free and then you won't have to do .Any() later which causes to iterate over? Also, does it need to be IEnumerable? We're not iterating over the list aside from the "exists" check?

Reply:
Any does not iterate over, it returns right away if there is one element in the backing container. It doesn't need to be an IEnumerable but there is no downsides in this case.